test/Platforms/Echo.Platforms.AsmResolver.Tests/Emulation/CilVirtualMachineTest.cs

```diff
@@ -662,5 +662,59 @@ public void StepPrefixedInstructionsShouldStepOverAllInstructions()
             _mainThread.Step();
             Assert.Equal(instructions[3].Offset, frame.ProgramCounter);
         }
+        
+        [Fact]
+        public void CallDelegate()
```

Comment:
Is there a way to clean up this test a bit, such that it is a bit more clear what exactly is tested for? I am having a bit of a hard time understanding exactly what is happening in this test. 

You can be explicit and repetitive in unit tests. The more explicit they are (especially in their asserts) the better context it gives us when something breaks in the future.

Reply:
I think it's clearer now.